services/motion/builtin/move_request.go

```diff
@@ -304,6 +291,22 @@ func (mr *moveRequest) obstaclesIntersectPlan(
 			inputMap := referenceframe.StartPositions(mr.planRequest.FrameSystem)
 			inputMap[mr.kinematicBase.Name().ShortName()] = currentInputs
 
+			// get the current position of the base
+			currentPosition, err := mr.kinematicBase.CurrentPosition(ctx)
+			if err != nil {
+				return state.ExecuteResponse{}, err
+			}
+
+			// Note: the value of wayPointIndex is subject to change between when this function is first entered
+			// versus when CheckPlan is actually called.
+			// We load the wayPointIndex value to ensure that all information is up to date.
+			waypointIndex := int(mr.waypointIndex.Load())
+
+			remainingPlan, err := motionplan.RemainingPlan(plan, waypointIndex-1)
```

Comment:
will `waypointIndex` ever be 1?

Reply:
Yes.

Suppose we have constructed some plan with the following inputs `[ [0, 0, 0], [a, b, c], [d, e, f], [0, 0, 0] ]`.

`waypointIndex := int(mr.waypointIndex.Load())` can be `1` if we call `obstaclesIntersectPlan` before we have finished executing `[a, b, c]`.

Reply:
I'm sorry I meant to write 0.  XD.  I'm trying to check if we're going to have array index problems here

Reply:
ahh I gotchu, I believe this is impossible ™️ . This is my motivation for thinking so:
https://github.com/viamrobotics/rdk/blob/main/services/motion/builtin/move_request.go#L755

Reply:
I am convinced.  Thanks!